File: packages/editor-schema/src/__tests__/schema.test.ts
import { describe, expect, it } from "vitest";
import {
  BLOCK_NODE_NAMES,
  INLINE_NODE_NAMES,
  MARK_NAMES,
  SCHEMA_VERSION,
  emptyDoc,
  isDocRoot,
} from "../schema.js";

describe("schema catalogue", () => {
  it("freezes the block-level node set", () => {
    expect([...BLOCK_NODE_NAMES]).toEqual([
      "paragraph",
      "heading",
      "bullet_list",
      "ordered_list",
      "check_list",
      "code_block",
      "image",
      "attachment",
    ]);
  });

  it("freezes the inline node set", () => {
    expect([...INLINE_NODE_NAMES]).toEqual(["text", "code_inline", "internal_link", "hard_break"]);
  });

  it("freezes the mark set", () => {
    expect([...MARK_NAMES]).toEqual(["bold", "italic", "strike"]);
  });

  it("pins SCHEMA_VERSION to 1", () => {
    expect(SCHEMA_VERSION).toBe(1);
  });
});

describe("emptyDoc", () => {
  it("returns a doc with a single empty paragraph", () => {
    const doc = emptyDoc();
    expect(isDocRoot(doc)).toBe(true);
    expect(doc.content).toEqual([{ type: "paragraph" }]);
  });
});

describe("isDocRoot", () => {
  it("rejects non-doc shapes", () => {
    expect(isDocRoot(null)).toBe(false);
    expect(isDocRoot({ type: "paragraph", content: [] })).toBe(false);
    expect(isDocRoot({ type: "doc" })).toBe(false);
  });
});

File: packages/android/editor-schema/src/test/kotlin/app/notes/editor/SchemaTest.kt
package app.notes.editor

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class SchemaTest {
    @Test
    fun `block node catalogue is frozen`() {
        assertEquals(
            listOf("paragraph", "heading", "bullet_list", "ordered_list", "check_list", "code_block", "image", "attachment"),
            BlockNodeName.entries.map { it.wireName },
        )
    }

    @Test
    fun `inline node catalogue is frozen`() {
        assertEquals(
            listOf("text", "code_inline", "internal_link", "hard_break"),
            InlineNodeName.entries.map { it.wireName },
        )
    }

    @Test
    fun `mark catalogue is frozen`() {
        assertEquals(listOf("bold", "italic", "strike"), MarkName.entries.map { it.wireName })
    }

    @Test
    fun `SCHEMA_VERSION matches the TS twin`() {
        assertEquals(1, SCHEMA_VERSION)
    }

    @Test
    fun `Doc empty has a single paragraph`() {
        val doc = Doc.empty()
        assertEquals(1, doc.content.size)
        assertEquals("paragraph", doc.content.first().type)
    }
}
